### PR TITLE
KAFKA-3694: Ensure broker Zk deregistration prior to restart in ReplicationTest

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

**Comment by ijuma** [comment]
LGTM. I'll let @granders and/or @ewencp take a look too since they're more familiar with the test itself.

**Comment by hachikuji** [comment]
Tests passing here: https://jenkins.confluent.io/view/All/job/system-test-kafka-branch-builder/443/console

File: tests/kafkatest/tests/core/replication_test.py [hunk 1]
```diff
@@ -65,15 +65,12 @@ def hard_bounce(test, broker_type):
...
-        # zookeeper and the broker cluster have registered the loss of the leader/controller.
-        # Waiting for a new leader for the topic-partition/controller to be elected is a reasonable heuristic for this.
+        # zookeeper has registered the loss by expiring the broker's session timeout.
 
-        def role_reassigned():
-            current_elected_broker = broker_node(test, broker_type)
-            return current_elected_broker is not None and current_elected_broker != prev_broker_node
+        wait_until(lambda: len(test.kafka.pids(prev_broker_node)) == 0 and not test.kafka.is_registered(prev_broker_node),
+                   timeout_sec=test.kafka.zk_session_timeout + 5,
+                   err_msg="Failed to see timely deregistration of hard-killed broker")
```
Comment (granders) [inline tests/kafkatest/tests/core/replication_test.py]: Include node info in the error message?

**Comment by granders** [comment]
@hachikuji One minor note, but LGTM
Also, replication tests are look healthy in this branch: 
http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2016-05-11--001.1462947131--hachikuji--KAFKA-3694--50630af/report.html

**Comment by hachikuji** [comment]
@granders Thanks for taking a look. I've updated the error message.

**Comment by ijuma** [comment]
LGTM, merging to trunk and 0.10.0.

**Pull request closed without merging**